src/graphql/fragments/actions.graphql

```diff
@@ -8,10 +8,28 @@ fragment ColonyAction on ColonyAction {
   initiatorUser {
     ...User
   }
+  initiatorColony {
+    ...Colony
+  }
+  initiatorExtension {
+    ...Extension
+  }
+  initiatorToken {
+    ...Token
+  }
```

Comment:
Do we really need to query all the fields of those, or should a more restricted fragments be created? Getting the `Colony` fragment after the `balances` were included already took a performance hit and I feel like this would slow down the action query quite a bit if any of these are not null

Reply:
Good point. I think we need Fragments for these four (User, Colony, Extension, Token) in the "friendly name" format. Ie: just `name`, `displayName`, `address`

Reply:
This is still outstanding I believe and potentially more urgent than my other comments